### ISSUE TITLE
[Bug]: Data Source: linode_instance_types not filtering on class properly

### ISSUE THREAD

**Issue by elreydetoda** [linode/terraform-provider-linode#703]
### Terraform Version

1.3.1

### Linode Provider Version

v1.29.2

### Effected Terraform Resources

[Data Source: linode_instance_types](https://registry.terraform.io/providers/linode/linode/latest/docs/data-sources/instance_types#class)

### Terraform Config Files

```terraform
provider "linode" {}
data "linode_instance_types" "instance-options" {
  filter {
    name = "vcpus"
    values = [ "4" ]
  }
  filter {
    name = "memory"
    values = [ "8192" ]
  }
  filter {
    name = "label"
    values = [ "Linode" ]
    match_by = "substring"
  }
}
output "instances-options" {
  value = data.linode_instance_types.instance-options
}
terraform {
  required_providers {
    linode = {
      source = "linode/linode"
    }
  }
}
```

All of my code I've used to validate the error is also here: https://gitlab.com/elreydetoda-infra/iac/misc/-/blob/06455eb25ec0326c61e2e1d8811fb2f49d4b5400/apps/jitsi/linode_discovery/main.tf

### Debug Output

```shell
$ for action in plan apply ; do ../../../scripts/local_terraform-helper.sh 'AWS_SECRET_ACCESS_KEY|AWS_ACCESS_KEY_ID|LINODE_TOKEN' "${action}|-no-color"; done 

[0m[1mInitializing the backend...[0m

[0m[1mInitializing provider plugins...[0m
- Reusing previous version of linode/linode from the dependency lock file
- Using previously-installed linode/linode v1.29.2

[0m[1m[32mTerraform has been successfully initialized![0m[32m[0m

No changes. Your infrastructure matches the configuration.

Terraform has compared your real infrastructure against your configuration and found no differences, so no changes are needed.

[0m[1mInitializing the backend...[0m

[0m[1mInitializing provider plugins...[0m
- Reusing previous version of linode/linode from the dependency lock file
- Using previously-installed linode/linode v1.29.2

[0m[1m[32mTerraform has been successfully initialized![0m[32m[0m

Apply complete! Resources: 0 added, 0 changed, 0 destroyed.

Outputs:

instances-options = {
  "filter" = tolist([
    {
      "match_by" = "exact"
      "name" = "vcpus"
      "values" = tolist([
        "4",
      ])
    },
    {
      "match_by" = "exact"
      "name" = "memory"
      "values" = tolist([
        "8192",
      ])
    },
    {
      "match_by" = "exact"
      "name" = "class"
      "values" = tolist([
        "standard",
      ])
    },
  ])
  "id" = "XXXX"
  "order" = "asc"
  "order_by" = tostring(null)
  "types" = tolist([])
}
```

### Expected Behavior

I expect the behavior to limit down the search results to only the standard class (currently am able to use a substring of the label to do the same), but the expected output is below.

```console
$ for action in plan apply ; do ../../../scripts/local_terraform-helper.sh 'AWS_SECRET_ACCESS_KEY|AWS_ACCESS_KEY_ID|LINODE_TOKEN' "${action}|-no-color"; done 
[0m[1mInitializing the backend...[0m

[0m[1mInitializing provider plugins...[0m
- Reusing previous version of linode/linode from the dependency lock file
- Using previously-installed linode/linode v1.29.2

[0m[1m[32mTerraform has been successfully initialized![0m[32m[0m

Changes to Outputs:
  ~ instances-options = {
      ~ filter   = [
            # (1 unchanged element hidden)
            {
                match_by = "exact"
                name     = "memory"
                values   = [
                    "8192",
                ]
            },
          + {
              + match_by = "substring"
              + name     = "label"
              + values   = [
                  + "Linode",
                ]
            },
        ]
      ~ id       = "XXXX" -> "XXXX"
      ~ types    = [
            {
                addons      = [
                    {
                        backups = [
                            {
                                price = [
                                    {
                                        hourly  = 0.014999999664723873
                                        monthly = 10
                                    },
                                ]
                            },
                        ]
                    },
                ]
                class       = "standard"
                disk        = 163840
                id          = "g6-standard-4"
                label       = "Linode 8GB"
                memory      = 8192
                network_out = 5000
                price       = [
                    {
                        hourly  = 0.05999999865889549
                        monthly = 40
                    },
                ]
                transfer    = 5000
                vcpus       = 4
            },
          - {
              - addons      = [
                  - {
                      - backups = [
                          - {
                              - price = [
                                  - {
                                      - hourly  = 0.014999999664723873
                                      - monthly = 10
                                    },
                                ]
                            },
                        ]
                    },
                ]
              - class       = "dedicated"
              - disk        = 163840
              - id          = "g6-dedicated-4"
              - label       = "Dedicated 8GB"
              - memory      = 8192
              - network_out = 5000
              - price       = [
                  - {
                      - hourly  = 0.09000000357627869
                      - monthly = 60
                    },
                ]
              - transfer    = 5000
              - vcpus       = 4
            },
        ]
        # (2 unchanged elements hidden)
    }

You can apply this plan to save these new output values to the Terraform state, without changing any real infrastructure.

[0m[1mInitializing the backend...[0m

[0m[1mInitializing provider plugins...[0m
- Reusing previous version of linode/linode from the dependency lock file
- Using previously-installed linode/linode v1.29.2

[0m[1m[32mTerraform has been successfully initialized![0m[32m[0m

Apply complete! Resources: 0 added, 0 changed, 0 destroyed.

Outputs:

instances-options = {
  "filter" = tolist([
    {
      "match_by" = "exact"
      "name" = "vcpus"
      "values" = tolist([
        "4",
      ])
    },
    {
      "match_by" = "exact"
      "name" = "memory"
      "values" = tolist([
        "8192",
      ])
    },
    {
      "match_by" = "substring"
      "name" = "label"
      "values" = tolist([
        "Linode",
      ])
    },
  ])
  "id" = "XXXX"
  "order" = "asc"
  "order_by" = tostring(null)
  "types" = tolist([
    {
      "addons" = tolist([
        {
          "backups" = tolist([
            {
              "price" = tolist([
                {
                  "hourly" = 0.014999999664723873
                  "monthly" = 10
                },
              ])
            },
          ])
        },
      ])
      "class" = "standard"
      "disk" = 163840
      "id" = "g6-standard-4"
      "label" = "Linode 8GB"
      "memory" = 8192
      "network_out" = 5000
      "price" = tolist([
        {
          "hourly" = 0.05999999865889549
          "monthly" = 40
        },
      ])
      "transfer" = 5000
      "vcpus" = 4
    },
  ])
}
```

These are the possible options to choose from (without the class filtering applied to show which I'm trying to filter):

```console
$ for action in plan apply ; do ../../../scripts/local_terraform-helper.sh 'AWS_SECRET_ACCESS_KEY|AWS_ACCESS_KEY_ID|LINODE_TOKEN' "${action}|-no-color"; done 


[0m[1mInitializing the backend...[0m

[0m[1mInitializing provider plugins...[0m
- Reusing previous version of linode/linode from the dependency lock file
- Using previously-installed linode/linode v1.29.2

[0m[1m[32mTerraform has been successfully initialized![0m[32m[0m

Changes to Outputs:
  ~ instances-options = {
      ~ filter   = [
            # (1 unchanged element hidden)
            {
                match_by = "exact"
                name     = "memory"
                values   = [
                    "8192",
                ]
            },
          - {
              - match_by = "exact"
              - name     = "class"
              - values   = [
                  - "standard",
                ]
            },
        ]
      ~ id       = "XXXX" -> "XXXX"
      ~ types    = [
          + {
              + addons      = [
                  + {
                      + backups = [
                          + {
                              + price = [
                                  + {
                                      + hourly  = 0.014999999664723873
                                      + monthly = 10
                                    },
                                ]
                            },
                        ]
                    },
                ]
              + class       = "standard"
              + disk        = 163840
              + id          = "g6-standard-4"
              + label       = "Linode 8GB"
              + memory      = 8192
              + network_out = 5000
              + price       = [
                  + {
                      + hourly  = 0.05999999865889549
                      + monthly = 40
                    },
                ]
              + transfer    = 5000
              + vcpus       = 4
            },
          + {
              + addons      = [
                  + {
                      + backups = [
                          + {
                              + price = [
                                  + {
                                      + hourly  = 0.014999999664723873
                                      + monthly = 10
                                    },
                                ]
                            },
                        ]
                    },
                ]
              + class       = "dedicated"
              + disk        = 163840
              + id          = "g6-dedicated-4"
              + label       = "Dedicated 8GB"
              + memory      = 8192
              + network_out = 5000
              + price       = [
                  + {
                      + hourly  = 0.09000000357627869
                      + monthly = 60
                    },
                ]
              + transfer    = 5000
              + vcpus       = 4
            },
        ]
        # (2 unchanged elements hidden)
    }

You can apply this plan to save these new output values to the Terraform state, without changing any real infrastructure.

[0m[1mInitializing the backend...[0m

[0m[1mInitializing provider plugins...[0m
- Reusing previous version of linode/linode from the dependency lock file
- Using previously-installed linode/linode v1.29.2

[0m[1m[32mTerraform has been successfully initialized![0m[32m[0m

Apply complete! Resources: 0 added, 0 changed, 0 destroyed.

Outputs:

instances-options = {
  "filter" = tolist([
    {
      "match_by" = "exact"
      "name" = "vcpus"
      "values" = tolist([
        "4",
      ])
    },
    {
      "match_by" = "exact"
      "name" = "memory"
      "values" = tolist([
        "8192",
      ])
    },
  ])
  "id" = "XXXX"
  "order" = "asc"
  "order_by" = tostring(null)
  "types" = tolist([
    {
      "addons" = tolist([
        {
          "backups" = tolist([
            {
              "price" = tolist([
                {
                  "hourly" = 0.014999999664723873
                  "monthly" = 10
                },
              ])
            },
          ])
        },
      ])
      "class" = "standard"
      "disk" = 163840
      "id" = "g6-standard-4"
      "label" = "Linode 8GB"
      "memory" = 8192
      "network_out" = 5000
      "price" = tolist([
        {
          "hourly" = 0.05999999865889549
          "monthly" = 40
        },
      ])
      "transfer" = 5000
      "vcpus" = 4
    },
    {
      "addons" = tolist([
        {
          "backups" = tolist([
            {
              "price" = tolist([
                {
                  "hourly" = 0.014999999664723873
                  "monthly" = 10
                },
              ])
            },
          ])
        },
      ])
      "class" = "dedicated"
      "disk" = 163840
      "id" = "g6-dedicated-4"
      "label" = "Dedicated 8GB"
      "memory" = 8192
      "network_out" = 5000
      "price" = tolist([
        {
          "hourly" = 0.09000000357627869
          "monthly" = 60
        },
      ])
      "transfer" = 5000
      "vcpus" = 4
    },
  ])
}

```

### Actual Behavior

Currently it's not returning any items based on my query:

```console
$ for action in plan apply ; do ../../../scripts/local_terraform-helper.sh 'AWS_SECRET_ACCESS_KEY|AWS_ACCESS_KEY_ID|LINODE_TOKEN' "${action}|-no-color"; done


[0m[1mInitializing the backend...[0m

[0m[1mInitializing provider plugins...[0m
- Reusing previous version of linode/linode from the dependency lock file
- Using previously-installed linode/linode v1.29.2

[0m[1m[32mTerraform has been successfully initialized![0m[32m[0m

No changes. Your infrastructure matches the configuration.

Terraform has compared your real infrastructure against your configuration and found no differences, so no changes are needed.

[0m[1mInitializing the backend...[0m

[0m[1mInitializing provider plugins...[0m
- Reusing previous version of linode/linode from the dependency lock file
- Using previously-installed linode/linode v1.29.2

[0m[1m[32mTerraform has been successfully initialized![0m[32m[0m

Apply complete! Resources: 0 added, 0 changed, 0 destroyed.

Outputs:

instances-options = {
  "filter" = tolist([
    {
      "match_by" = "exact"
      "name" = "vcpus"
      "values" = tolist([
        "4",
      ])
    },
    {
      "match_by" = "exact"
      "name" = "memory"
      "values" = tolist([
        "8192",
      ])
    },
    {
      "match_by" = "exact"
      "name" = "class"
      "values" = tolist([
        "standard",
      ])
    },
  ])
  "id" = "XXXX"
  "order" = "asc"
  "order_by" = tostring(null)
  "types" = tolist([])
}
```

### Steps to Reproduce

Everything is listed here: https://gitlab.com/elreydetoda-infra/iac/misc/-/blob/06455eb25ec0326c61e2e1d8811fb2f49d4b5400/apps/jitsi/linode_discovery/

only thing I've done is use a container to host the terraform binary (made by Gitlab), and then pass in my API keys as environment variables via my helper script (show above in the console output).

**Comment by LBGarber** [comment]
Thanks for the bug report!

I've put up a PR to fix this issue (#704) and this issue should be resolved in our next release. The details of the fix can be found in the issue description.

Let me know if you run into any other issues :slightly_smiling_face:

**Comment by elreydetoda** [comment]
Awesome, thanks @LBGarber!
